packages/react/src/components/IconButton/index.tsx

```diff
@@ -1,52 +1,50 @@
-import React, { useRef } from 'react';
-import PropTypes from 'prop-types';
+import React, { useRef, forwardRef } from 'react';
 import classnames from 'classnames';
 import { Placement } from '@popperjs/core';
 import Icon from '../Icon';
 import Tooltip from '../Tooltip';
-
 export interface IconButtonProps
   extends React.ButtonHTMLAttributes<HTMLButtonElement> {
   icon: string;
   label: string;
   tooltipPlacement?: Placement;
 }
-
-export default function IconButton({
-  icon,
-  label,
-  tooltipPlacement = 'auto',
-  className,
-  ...other
-}: IconButtonProps) {
-  const buttonRef = useRef<HTMLButtonElement>(null);
-  return (
-    <React.Fragment>
-      <button
-        type={'button'}
-        className={classnames('IconButton', className)}
-        ref={buttonRef}
-        {...other}
-      >
-        <Icon type={icon} />
-      </button>
-      <Tooltip
-        target={buttonRef}
-        placement={tooltipPlacement}
-        association="aria-labelledby"
-        hideElementOnHidden
-      >
-        {label}
-      </Tooltip>
-    </React.Fragment>
-  );
-}
-
+const IconButton = forwardRef<HTMLButtonElement, IconButtonProps>(
+  (
+    {
+      icon,
+      label,
+      tooltipPlacement = 'auto',
+      className,
+      ...other
+    }: IconButtonProps,
+    ref
+  ): JSX.Element => {
+    const buttonRef = useRef<HTMLButtonElement>(null);
+    if (typeof ref === 'function') {
+      ref(buttonRef.current);
+    }
```

Comment:
I think we may want to use [`useImperitiveHandle`](https://reactjs.org/docs/hooks-reference.html#useimperativehandle) here:

```jsx
useImperitiveHandle(ref, () => buttonRef.current)
```

That way the `ref` passed as props below can continue to use the combined ref value, whether it was passed from the parent as a ref or functional ref, or not.

Reply:
What is the difference between the imperativeHandle and what is implemented in the PR? The TS compiler really REALLY does not like: ```useImperativeHandle(ref, () => buttonRef.current)```

Reply:
From what i see `useImperativeHandle` is more suited to provide a subset of (in this case ) <HTMLButton> properties to the forwardRef like "focus". I would think we want to expose the full ref, which is what is implemented.

Reply:
ah, nevermind... i got this to compile, changes inbound and down.

Reply:
FWIW, I was able to get it working with this as well:

```tsx
function IconButton({
  icon,
  label,
  tooltipPlacement = 'auto',
  className,
  ...other
}: IconButtonProps, ref: React.Ref<HTMLButtonElement>) {
  const buttonRef = useRef<HTMLButtonElement>(null);
  useImperativeHandle(ref, () => buttonRef.current as HTMLButtonElement)
  return (...)
}
```

What you have is probably fine too.